File: frontend/src/modules/reparaciones/graphql/queries.js
import { gql } from '@apollo/client'

export const GET_TIPOS_REPARACION = gql`
  query {
    tiposReparacion {
      idTipoReparacion
      nombre
      descripcion
      estado
    }
  }
`

export const GET_TIPOS_DETALLE = gql`
  query {
    tiposDetalle {
      idTipo
      nombre
      descripcion
    }
  }
`

export const GET_EQUIPOS = gql`
  query {
    equipos {
      idEquipo
      nombre
      modelo
      descripcion
      tipoEquipo
      serieImei
      fechaIngreso
      estado
      idCliente { idCliente nombreCompleto }
    }
  }
`

export const GET_NOTAS_REPARACION = gql`
  query {
    notasReparacion {
      idNotaReparacion
      fecha
      fallaReportada
      fallaReal
      fechaEntrega
      monto
      tipoPago
      estado
      idEquipo { idEquipo nombre modelo }
      idCliente { idCliente nombreCompleto }
      idUsuario { username }
      idTipoReparacion { nombre }
    }
  }
`

export const GET_DIAGNOSTICOS = gql`
  query {
    diagnosticos {
      idDiagnostico
      descripcion
      causaRaiz
      tiempoEstimadoHrs
      fechaDiagnostico
      idNotaReparacion {
        idNotaReparacion
        estado
        idEquipo { nombre modelo }
        idCliente { nombreCompleto }
      }
    }
  }
`

export const GET_COTIZACIONES = gql`
  query {
    cotizaciones {
      idCotizacion
      costoRepuesto
      manoObra
      total
      estado
      idDiagnostico {
        idDiagnostico
        idNotaReparacion {
          idNotaReparacion
          idEquipo { nombre modelo }
          idCliente { nombreCompleto }
        }
      }
      detalles {
        idDetalleReparacion
        descripcion
        cantidad
        precioUnitario
        precioTotal
        idTipoDetalle { nombre }
        idProducto { nombre }
      }
    }
  }
`

File: backend/reparaciones/schema/queries.py
import graphene
from graphene_django import DjangoObjectType
from reparaciones.models import (
    TipoReparacion, TipoDetalle, Equipo,
    NotaReparacion, Diagnostico, CotizacionReparacion, DetalleReparacion
)


class TipoReparacionType(DjangoObjectType):
    class Meta:
        model = TipoReparacion
        fields = ('id_tipo_reparacion', 'nombre', 'descripcion', 'estado')


class TipoDetalleType(DjangoObjectType):
    class Meta:
        model = TipoDetalle
        fields = ('id_tipo', 'nombre', 'descripcion')


class EquipoType(DjangoObjectType):
    class Meta:
        model = Equipo
        fields = ('id_equipo', 'nombre', 'modelo', 'descripcion',
                  'tipo_equipo', 'serie_imei', 'fecha_ingreso', 'estado', 'id_cliente')


class NotaReparacionType(DjangoObjectType):
    class Meta:
        model = NotaReparacion
        fields = ('id_nota_reparacion', 'fecha', 'falla_reportada', 'falla_real',
                  'fecha_entrega', 'monto', 'tipo_pago', 'estado',
                  'id_equipo', 'id_cliente', 'id_usuario', 'id_tipo_reparacion')


class DiagnosticoType(DjangoObjectType):
    class Meta:
        model = Diagnostico
        fields = ('id_diagnostico', 'descripcion', 'causa_raiz',
                  'tiempo_estimado_hrs', 'fecha_diagnostico', 'id_nota_reparacion')


class CotizacionReparacionType(DjangoObjectType):
    class Meta:
        model = CotizacionReparacion
        fields = ('id_cotizacion', 'costo_repuesto', 'mano_obra', 'total',
                  'estado', 'id_diagnostico', 'detalles')


class DetalleReparacionType(DjangoObjectType):
    class Meta:
        model = DetalleReparacion
        fields = ('id_detalle_reparacion', 'descripcion', 'observaciones',
                  'cantidad', 'precio_unitario', 'precio_total',
                  'id_cotizacion', 'id_tipo_detalle', 'id_producto')


class Query(graphene.ObjectType):
    tipos_reparacion    = graphene.List(TipoReparacionType)
    tipos_detalle       = graphene.List(TipoDetalleType)
    equipos             = graphene.List(EquipoType)
    equipo              = graphene.Field(EquipoType, id_equipo=graphene.Int(required=True))
    notas_reparacion    = graphene.List(NotaReparacionType)
    nota_reparacion     = graphene.Field(NotaReparacionType, id_nota_reparacion=graphene.Int(required=True))
    diagnosticos        = graphene.List(DiagnosticoType)
    cotizaciones        = graphene.List(CotizacionReparacionType)
    cotizacion          = graphene.Field(CotizacionReparacionType, id_cotizacion=graphene.Int(required=True))

    def resolve_tipos_reparacion(root, info):
        return TipoReparacion.objects.filter(estado='activo')

    def resolve_tipos_detalle(root, info):
        return TipoDetalle.objects.all()

    def resolve_equipos(root, info):
        return Equipo.objects.select_related('id_cliente').all()

    def resolve_equipo(root, info, id_equipo):
        try:
            return Equipo.objects.get(pk=id_equipo)
        except Equipo.DoesNotExist:
            return None

    def resolve_notas_reparacion(root, info):
        return NotaReparacion.objects.select_related(
            'id_equipo', 'id_cliente', 'id_usuario', 'id_tipo_reparacion'
        ).all()

    def resolve_nota_reparacion(root, info, id_nota_reparacion):
        try:
            return NotaReparacion.objects.get(pk=id_nota_reparacion)
        except NotaReparacion.DoesNotExist:
            return None

    def resolve_diagnosticos(root, info):
        return Diagnostico.objects.select_related('id_nota_reparacion').all()

    def resolve_cotizaciones(root, info):
        return CotizacionReparacion.objects.select_related('id_diagnostico').prefetch_related('detalles').all()

    def resolve_cotizacion(root, info, id_cotizacion):
        try:
            return CotizacionReparacion.objects.get(pk=id_cotizacion)
        except CotizacionReparacion.DoesNotExist:
            return None